src/app/WriteHandler.h

```diff
@@ -175,6 +180,8 @@ class WriteHandler : public Messaging::ExchangeDelegate
     //  Where (1)-(3) will be consistent among the whole list write request, while (4) and (5) are not appliable to group writes.
     bool mAttributeWriteSuccessful                = false;
     Optional<AttributeAccessToken> mACLCheckCache = NullOptional;
+
+    InteractionModelEngine * mpImEngine = nullptr;
```

Comment:
@bzbarsky-apple I think we should not increase the memory in writer handler and read handler? any better suggestion on this effort?

Reply:
WriteHandler and TimedHandler  are mainly used in embedded device with low memory, what is the benefit to move  away from singleton toward pointer?

Reply:
https://github.com/project-chip/connectedhomeip/pull/31494#discussion_r1467190141, As Boris noted, we're already exceeding our memory target. Instead of adding more, we need to find ways to reduce our memory usage.  These handlers are meant for embedded devices with tight memory constraints. Adding more pointers for each handler could significantly increase our memory footprint.  I don't believe the testing benefits outweigh the potential memory issues. @andreilitvin

Reply:
The most benefit for it would be able to have our own instance of InteractionModelEngine instead of the global one and having multiple InteractionModelEngines if needed. I guess there can be a way to avoid it my introducing a compilation flag to determine using a pointer or a singleton from @mspang 's idea. That will make things ugly but what do you think?

Reply:
I would pay the 150 bytes of flash if this shows a unit test capability that was not there before. Could we make sure we show that functionality in this same PR?

Otherwise we just see the increase and not the benefit.